tests/zfs-tests/tests/functional/block_cloning/block_cloning.kshlib

```diff
@@ -28,10 +28,7 @@
 
 function have_same_content
 {
-	typeset hash1=$(md5digest $1)
-	typeset hash2=$(md5digest $2)
-
-	log_must [ "$hash1" = "$hash2" ]
+	log_must cmp -s $1 $2
```

Comment:
Why the change here?   The `md5digest` wrapper in `tests/zfs-tests/include/libtest.shlib` is already written to be portable, and this change seems to be causing some test failures on Linux.

Reply:
> Why the change here? The `md5digest` wrapper in `tests/zfs-tests/include/libtest.shlib` is already written to be portable, and this change seems to be causing some test failures on Linux.

That was just a microoptimization, as we don't need to calculate checksum to compare content of two files. From what I checked cmp(1) is available on Linux and I haven't seen any tests failing because of that, but no problem, I can revert that.

Thanks for mentioning that, BTW, as in my bclone tests I was using openssl sha256 and now I know we also have sha256digest:)